probe-rs/src/architecture/arm/memory/adi_v5_memory_interface.rs

```diff
@@ -392,32 +462,98 @@ where
         Ok(())
     }
 
+    /// Read a block of 8 bit words at `address`.
+    ///
+    /// The number of words read is `data.len()`.
     pub fn read_8(
         &mut self,
         access_port: MemoryAp,
         address: u64,
         data: &mut [u8],
     ) -> Result<(), AccessPortError> {
+        if self.supports_only_32bit_data_size {
+            return Err(AccessPortError::UnsupportedTransferWidth(8));
+        }
+
         if data.is_empty() {
             return Ok(());
         }
 
-        let aligned = aligned_range(address, data.len())?;
-        let aligned_len = (aligned.end - aligned.start) as usize;
+        let start_address = address;
+        let mut data_u32 = vec![0u32; data.len()];
+
+        // Second we read in 32 bit reads until we have less than 32 bits left to read.
+        let csw = self.build_csw_register(DataSize::U8);
+        self.write_csw_register(access_port, csw)?;
+
+        let mut address = address;
+        self.write_tar_register(access_port, address)?;
+
+        // The maximum chunk size we can read before data overflows.
+        let max_chunk_size_bytes = 0x400;
```

Comment:
Is there an explanation why this is the maximum chunk size?

Reply:
I think this was introduced by you and is a limitation by the ARM spec. I will add a comment.